doc/developers/maintainer.rst

```diff
@@ -130,44 +128,20 @@ Making a release
    candidate period, the latest stable is two versions behind the master
    branch, instead of one.
 
-3. At this point all relevant PRs should have been merged into the `0.99.X`
-   branch. Create the source tarball:
-
-   - Wipe clean your repo:
-
-     .. prompt:: bash $
-
-       git clean -xfd
-
-   - Generate the tarball:
-
-     .. prompt:: bash $
-
-       python setup.py sdist
-
-   - You can also test a binary dist build using:
-
-     .. prompt:: bash $
-
-       python setup.py bdist_wheel
-
-   - You can test if PyPi is going to accept the package using:
-
-     .. prompt:: bash $
-
-       twine check dist/*
+3. Proceed with caution. Ideally, tags should be created when you're almost
+   certain that the release is ready, since adding a tag to the main repo can
+   trigger certain automated processes. You can create a PR in the main repo
+   and trigger the wheel builder with the ``[cd build]`` commit marker using
+   the command:
+   
+   .. prompt:: bash $
 
-   You can run ``twine check`` after step 5 (fetching artifacts) as well.
+    git commit --allow-empty -m "Trigger wheel builder workflow: [cd build]"
 
-   The result should be in the `dist/` folder. We will upload it later
-   with the wheels. Check that you can install it in a new virtualenv and
-   that the tests pass.
+   The wheel building workflow is managed by GitHub Actions and the results be browsed at:
+   https://github.com/scikit-learn/scikit-learn/actions?query=workflow%3A%22Wheel+builder%22
 
-4. Proceed with caution. Ideally, tags should be created when you're almost
-   certain that the release is ready, since adding a tag to the main repo can
-   trigger certain automated processes. You can test upload the ``sdist`` to
-   ``test.pypi.org``, and test the next step by setting ``BUILD_COMMIT`` to the
-   branch name (``0.99.X`` for instance) in a PR to the wheel building repo.
+   Then, you can upload the wheels and ``sdist`` to ``test.pypi.org`` for testing.
```

Comment:
I think this is suggesting to download the artifacts from github actions and try uploading it to `test.pypi.org`?

We can keep this in for now, but I would prefer to not have optional language in the checklist.

Reply:
> I think this is suggesting to download the artifacts from github actions and try uploading it to `test.pypi.org`?

That is. Although this would not be necessary when #18800 gets merged.

> We can keep this in for now, but I would prefer to not have optional language in the checklist.

What do you mean by "optional language in the checklist"?

Reply:
> What do you mean by "optional language in the checklist"?

Either we make strict recommendations or we don't make the recommendation in the first place.

Actually I think we should rather recommend to use wheelhouse uploader to collect the artifacts from the staging area
 and then give the command line to try to upload the to test.pypi.org first, and then if all goes well, upload to final pypi.org.

Reply:
But indeed this will be simplified once we use github actions to upload to  test.pypi.org / pypi.org.

Let's remove this from the steps for now.